src/yosys_mau/task_loop/context.py

```diff
@@ -205,6 +205,7 @@ def __getitem__(self, key: K) -> V:
             else:
                 if isinstance(value, _MISSING_TYPE):
                     raise KeyError(repr(key))
+                return value
```

Comment:
Nit, optional: I'd prefer fixes for functionality not added in a PR to be a separate commit as that makes things like bisection and cherry picking easier. (I don't think I'm very good at consistently following my own advice regarding this, though, and especially with mau where there's not too much concurrently ongoing development it's not really an issue.)

Reply:
I had to add this to prevent the log handler from hanging when it tried to access the default level, but that may not be necessary with the switch to task context dict.  I'll try remove this and see if it works.

Reply:
From the surrounding code it looks to me like your change fixes an actual bug and I wasn't suggesting to remove the fix from the PR but rather to have a separate commit containing just the fix, still included in this PR, so that the commit log makes it clear that a bug was fixed.

Although if you're less certain about whether it's an actual bug it's probably a good idea for me to also double check how this was intended to interact with the overall implementation. And maybe also add a regression test that covers that specific case, as, no matter which variant is correct, it's slightly concerning that all existing tests seem pass with both variants.